pylint/checkers/unicode.py

```diff
@@ -218,7 +218,7 @@ def _normalize_codec_name(codec: str) -> str:
 
 def _remove_bom(encoded: bytes, encoding: str) -> bytes:
     """Remove the bom if given from a line."""
-    if not encoding.startswith("utf"):
+    if encoding not in UNICODE_BOMS:
         return encoded
     bom = UNICODE_BOMS[encoding]
```

Comment:
since all UNICODE_BOMS are utf..., it's redundant to check if it starts with utf. Just checking if encoding is in UNICODE is sufficient and prevents unsafe KeyError